cpp/csp/python/PyAdapterManager.cpp

```diff
@@ -68,6 +68,25 @@ class PyAdapterManager : public AdapterManager
 static PyObject * PyAdapterManager_PyObject_starttime( PyAdapterManager_PyObject * self ) { return toPython( self -> manager -> starttime() ); }
 static PyObject * PyAdapterManager_PyObject_endtime( PyAdapterManager_PyObject * self )   { return toPython( self -> manager -> endtime() ); }
 
+static PyObject * PyAdapterManager_PyObject_shutdown_engine( PyAdapterManager_PyObject * self, PyObject * args, PyObject * kwargs )
+{
+    CSP_BEGIN_METHOD;
+
+    PyObject * pyException;
+    if( !PyArg_ParseTuple( args, "O", &pyException ) )
```

Comment:
can we move this into a method on PyEngine that takes the pyExpection PyObject
no need to duplicate the logic